src/storage/gc_worker.rs

```diff
@@ -212,25 +254,102 @@ impl<E: Engine> GCRunner<E> {
         );
         Ok(())
     }
-}
 
-impl<E: Engine> Runnable<GCTask> for GCRunner<E> {
-    fn run(&mut self, mut task: GCTask) {
-        GC_GCTASK_COUNTER.inc();
+    fn unsafe_destroy_range(&self, _: &Context, start_key: &Key, end_key: &Key) -> Result<()> {
+        info!(
+            "destroying range start_key: {}, end_key: {}",
+            start_key, end_key
+        );
+
+        // TODO: Refine usage of errors
+
+        let local_storage = self.local_storage.as_ref().ok_or_else(|| {
+            let e: Error = box_err!("destroy range not supported: local_storage not set");
+            warn!("destroy range failed: {:?}", &e);
+            e
+        })?;
+
+        // Convert keys to RocksDB layer form
+        // TODO: Logic coupled with raftstore's implementation. Maybe better design is to do it in
+        // somewhere of the same layer with apply_worker.
+        let start_data_key = keys::data_key(start_key.as_encoded());
+        let end_data_key = keys::data_end_key(end_key.as_encoded());
+
+        // First, call delete_files_in_range to free as much disk space as possible
+        // TODO: Will LOCK_CF cause problem here?
+        for cf in ALL_CFS {
+            let cf_handle = get_cf_handle(local_storage, cf).unwrap();
+            local_storage
+                .delete_files_in_range_cf(cf_handle, &start_data_key, &end_data_key, false)
```

Comment:
I think you should not delete all files for one time when the truncated table is very large, it would make the disk very busy? You should test it.